src/frontend/Navigation/Stack/AppScreens.tsx

```diff
@@ -55,21 +55,29 @@ import {HomeTabs} from '../Tab';
 import {SaveTrackScreen} from '../../screens/SaveTrack/SaveTrackScreen';
 import {ObservationFields} from '../../screens/ObservationFields';
 import {LanguageSettings} from '../../screens/Settings/AppSettings/LanguageSettings';
+import {BottomSheetModalProvider} from '@gorhom/bottom-sheet';
 
 export const TAB_BAR_HEIGHT = 70;
 
 // **NOTE**: No hooks allowed here (this is not a component, it is a function
 // that returns a react element)
-export const createDefaultScreenGroup = (
-  intl: (title: MessageDescriptor) => string,
-) => (
+export const createDefaultScreenGroup = ({
+  intl,
+  openDrawer,
+}: {
+  intl: (title: MessageDescriptor) => string;
+  openDrawer: () => void;
+}) => (
   <RootStack.Group key="default">
     <RootStack.Screen
       name="Home"
       options={{headerShown: false}}
       children={() => (
         <SharedLocationContextProvider>
-          <HomeTabs />
+          {/* This provider needs to be inside the navigator as it needs access to navigator state (aka it cant just use the one at the app level) */}
+          <BottomSheetModalProvider>
+            <HomeTabs openDrawer={openDrawer} />
+          </BottomSheetModalProvider>
```

Comment:
not sure I understand this. the provider is actually only providing a couple of methods that can be used via the hooks provided by the library: https://github.com/gorhom/react-native-bottom-sheet/blob/v4.5.1/src/contexts/modal/external.ts#L3-L6

you already have the provider set up in `AppProviders`, which this component is used under, so you shouldn't need to set up another provider here.

Reply:
Im realizing this is an outdated comment. The actual reason is this the most elegant way to open the bottom sheet behind the drawer

When we are taking tracks, and the user opens the drawer, we want bottom sheet to open behind the drawer. But if the user was to receive an invite, we want the the bottom sheet to open above the drawer. So this allows for that to happen, as the tracks will use this provider which is a child of the drawer component. While the invites lives outside the home screen, so it will use the other provider, which is a parent to the drawer, so it will show up over the drawer.

Reply:
ah i see! thanks for explaining - makes a lot more sense. also confirms that i didn't understand how this library works 😅